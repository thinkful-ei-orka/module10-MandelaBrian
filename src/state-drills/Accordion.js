import React from 'react';

class Accordion extends React.Component {
    static defaultProps = {

     sections: [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
      ]
    }

    state = {
        expandedIndex: -1
    };

    handleButtonClick = (index) => {
        
        this.setState({
            expandedIndex: index
        })
        // this.buildButtons(event.currentTarget.id)
    }

    hasP = (index) => {
        if (index === this.state.expandedIndex){
            return <p></p>
        } else {
            return ""
        }    
    }

    renderButtons() {

        return this.props.sections.map((sections, index) => ( 
          <li><button id = {index} key={index} onClick={() => this.handleButtonClick(index)}>
            {sections.title}
        </button>{this.hasP(index)}</li>  )) 
      }

    render() {
        
        return (
            <ul>
                {this.renderButtons()}
                
            </ul>
        )
      }
    }
    
export default Accordion
