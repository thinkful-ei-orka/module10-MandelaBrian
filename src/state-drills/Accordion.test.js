import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Accordion from './Accordion'

describe(`Accordion Component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI empty', () => {
    expect(
      renderer.create(<Accordion sections ={[]}/>).toJSON()
    ).toMatchSnapshot()
  })
  it('renders the UI no sections active', () => {
    expect(
      renderer.create(<Accordion />).toJSON()
    ).toMatchSnapshot()
  })
  it('renders the UI opens a clicked section', () => {
    expect(
      renderer.create(<Accordion sections ={[]}/>).toJSON()
    ).toMatchSnapshot()
  })
  it('renders the UI only opens last when mutliple have been clicked', () => {
    expect(
      renderer.create(<Accordion sections ={[]}/>).toJSON()
    ).toMatchSnapshot()
  })
})