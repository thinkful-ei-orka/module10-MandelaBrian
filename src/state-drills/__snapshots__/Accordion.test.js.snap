// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component renders the UI empty 1`] = `<ul />`;

exports[`Accordion Component renders the UI no sections active 1`] = `
<ul>
  <li>
    <button
      id={0}
      onClick={[Function]}
    >
      Section 1
    </button>
    
  </li>
  <li>
    <button
      id={1}
      onClick={[Function]}
    >
      Section 2
    </button>
    
  </li>
  <li>
    <button
      id={2}
      onClick={[Function]}
    >
      Section 3
    </button>
    
  </li>
</ul>
`;

exports[`Accordion Component renders the UI only opens last when mutliple have been clicked 1`] = `<ul />`;

exports[`Accordion Component renders the UI opens a clicked section 1`] = `<ul />`;
